[rpc] polish examples in docstrings and update docs to reflect correct use of
rpc.init_rpc

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30052 [rpc] polish examples in docstrings and update docs to reflect correct use of rpc.init_rpc**

Some of the examples provided in `rpc/api.py` were not updated along
with the code changes, this PR updates them. Also removes the
`dist.ProcessGroup` information since `init_rpc` now initializes a default
process group.

Differential Revision: [D18582596](https://our.internmc.facebook.com/intern/diff/D18582596/)

rohan-varma committed:
[rpc] polish examples in docstrings and update docs to reflect correct use of
rpc.init_rpc

Some of the examples provided in `rpc/api.py` were not updated along
with the code changes, this PR updates them. Also removes the
`dist.ProcessGroup` information since `init_rpc` now initializes a default
process group.

Differential Revision: [D18582596](https://our.internmc.facebook.com/intern/diff/D18582596/)

[ghstack-poisoned]

rohan-varma commented on docs/source/rpc.rst:
Basically, I'm trying to express that `init_rpc` runs `dist.init_process_group(backend=GLOO)` behind the scenes, is there a better way to write this?

pritamdamania87 commented on docs/source/rpc.rst:
Rewrite this as follows to make it clear that the default is ProcessGroup but other backends be available:

```
By default, this will initialize the `ProcessGroup` backend. The `ProcessGroup` backend internally uses gloo for communication
```

pritamdamania87 commented on docs/source/rpc.rst:
Use :ref: for this link instead of the full path.

pritamdamania87 reviewed (approved):
Looks good, although please address the comments before landing.

rohan-varma committed:
Update on "[rpc] polish examples in docstrings and update docs to reflect correct use of
rpc.init_rpc"

rpc.init_rpc**
* #30050 [rpc] rename join_rpc to wait_all_workers in public api

rpc.init_rpc

Some of the examples provided in `rpc/api.py` were not updated along
with the code changes, this PR updates them. Also removes the
`dist.ProcessGroup` information since `init_rpc` now initializes a default
process group.

Differential Revision: [D18582596](https://our.internmc.facebook.com/intern/diff/D18582596/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] polish examples in docstrings and update docs to reflect correct use of
rpc.init_rpc"

rpc.init_rpc**
* #30050 [rpc] rename join_rpc to wait_all_workers in public api
rpc.init_rpc**
* #30050 [rpc] rename join_rpc to wait_all_workers in public api

rpc.init_rpc

Some of the examples provided in `rpc/api.py` were not updated along
with the code changes, this PR updates them. Also removes the
`dist.ProcessGroup` information since `init_rpc` now initializes a default
process group.

Differential Revision: [D18582596](https://our.internmc.facebook.com/intern/diff/D18582596/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] polish examples in docstrings and update docs to reflect correct use of
rpc.init_rpc"

rpc.init_rpc**
* #30050 [rpc] rename join_rpc to wait_all_workers in public api
rpc.init_rpc**
* #30050 [rpc] rename join_rpc to wait_all_workers in public api
rpc.init_rpc**
* #30050 [rpc] rename join_rpc to wait_all_workers in public api

rpc.init_rpc

Some of the examples provided in `rpc/api.py` were not updated along
with the code changes, this PR updates them. Also removes the
`dist.ProcessGroup` information since `init_rpc` now initializes a default
process group.

Differential Revision: [D18582596](https://our.internmc.facebook.com/intern/diff/D18582596/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] polish examples in docstrings and update docs to reflect correct use of
rpc.init_rpc"

rpc.init_rpc**
* #30050 [rpc] rename join_rpc to wait_all_workers in public api
rpc.init_rpc**
* #30050 [rpc] rename join_rpc to wait_all_workers in public api
rpc.init_rpc**
* #30050 [rpc] rename join_rpc to wait_all_workers in public api
rpc.init_rpc**
* #30050 [rpc] rename join_rpc to wait_all_workers in public api

rpc.init_rpc

Some of the examples provided in `rpc/api.py` were not updated along
with the code changes, this PR updates them. Also removes the
`dist.ProcessGroup` information since `init_rpc` now initializes a default
process group.

Differential Revision: [D18582596](https://our.internmc.facebook.com/intern/diff/D18582596/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] polish examples in docstrings and update docs to reflect correct use of
rpc.init_rpc"

rpc.init_rpc**
* #30050 [rpc] rename join_rpc to wait_all_workers in public api
rpc.init_rpc**
* #30050 [rpc] rename join_rpc to wait_all_workers in public api
rpc.init_rpc**
* #30050 [rpc] rename join_rpc to wait_all_workers in public api
rpc.init_rpc**
* #30050 [rpc] rename join_rpc to wait_all_workers in public api
rpc.init_rpc**
* #30050 [rpc] rename join_rpc to wait_all_workers in public api

rpc.init_rpc

Some of the examples provided in `rpc/api.py` were not updated along
with the code changes, this PR updates them. Also removes the
`dist.ProcessGroup` information since `init_rpc` now initializes a default
process group.

Differential Revision: [D18582596](https://our.internmc.facebook.com/intern/diff/D18582596/)

[ghstack-poisoned]
